cyber/tools/cyber_monitor/general_channel_message.cc

```diff
@@ -89,9 +89,13 @@ double GeneralChannelMessage::frame_ratio(void) {
     if (old == 0) {
       return 0.0;
     }
+    int old_size = frame_size_sum_;
+    while (!frame_size_sum_.compare_exchange_strong(old_size, 0)) {
+    }
```

Comment:
Missing logic or intended waiting (very rare)?

Reply:
Expect to wait, as in the original code above:
    while (!frame_counter_.compare_exchange_strong(old, 0)) {
    }

Reply:
Will it cost too much CPU? I guess it's better to add something like `sleep(0.1s)` here.

Reply:
It's not necessary. The use of compare_exchange_strong is usually the same.
1) If frame_counter_is equal to old, frame_counter_is assigned 0 and returned true; 
2) if frame_counter_is changed elsewhere, frame_counter_is not equal to old, old is assigned frame_counter_ and returned false, the second loop will frame_counter_is equal to old, frame_counter_is assigned 0, returns true, and ends the loop. This is to complete frame_counter_= 0 atomically.

Reply:
If sleep (0.1) is added, frame_counter_is likely to be changed by other threads in 0.1s. In extreme cases, if frame_counter_in 0.1s is bound to be changed somewhere else, we will never be able to complete the operation of frame_counter_ = 0.

Reply:
http://www.cplusplus.com/reference/atomic/atomic/compare_exchange_strong/